File: projects/poly_mission_control/web_frontend/src/types/system.ts
export interface SystemMetrics {
  cpu_usage_pct: number;
  cpu_cores: number;
  cpu_model: string;
  load_avg_1m: number;
  load_avg_5m: number;
  load_avg_15m: number;
  cpu_cores_usage: number[];
  top_processes: {
    pid: string;
    name: string;
    cpu_pct: number;
    mem_pct: number;
  }[];
  memory_total_mb: number;
  memory_used_mb: number;
  memory_free_mb: number;
  disk_total_gb: number;
  disk_used_gb: number;
  disk_avail_gb: number;
  disk_used_pct: number;
  uptime_hours: number;
  crontab_active: boolean;
  telegram_bot_active: boolean;
  sidecar_active: boolean;
  last_batch_run?: string | null;
  [key: string]: any;
}

export interface GeminiAccount {
  index: number;
  name: string;
  is_active: boolean;
  is_cooling: boolean;
  cooldown_remaining_sec: number;
  status: string;
  token_health?: string;
  [key: string]: any;
}

export interface SubagentItem {
  id: string;
  name: string;
  role: string;
  status: 'idle' | 'running' | 'error';
  last_active?: string;
  current_task?: string;
  description?: string;
  task_count?: number;
  [key: string]: any;
}

export interface DaemonStatus {
  name: string;
  service_name: string;
  status: 'active' | 'inactive' | 'failed';
  uptime?: string;
  pid?: number;
  description?: string;
  is_running?: boolean;
  cpu_pct?: number;
  mem_mb?: number;
  [key: string]: any;
}

export interface CrontabJob {
  schedule: string;
  command: string;
  description?: string;
  last_run?: string;
  next_run?: string;
  log_path?: string;
  [key: string]: any;
}

export interface SkillItem {
  name: string;
  description: string;
  triggers: string[];
  path?: string;
  enabled?: boolean;
  title?: string;
  is_executable_present?: boolean;
  reference_path?: string;
  ref_lines?: number;
  [key: string]: any;
}

export interface WikiItem {
  name: string;
  path: string;
  title?: string;
  relative_path?: string;
  size_human?: string;
  size_bytes?: number;
  modified_time?: string;
  [key: string]: any;
}

File: projects/poly_mission_control/src/types/system.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProcessCpuInfo {
    pub pid: String,
    pub name: String,
    pub cpu_pct: f64,
    pub mem_pct: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SystemMetrics {
    pub cpu_usage_pct: f64,
    pub cpu_cores: usize,
    pub cpu_model: String,
    pub load_avg_1m: f64,
    pub load_avg_5m: f64,
    pub load_avg_15m: f64,
    pub cpu_cores_usage: Vec<f64>,
    pub top_processes: Vec<ProcessCpuInfo>,
    pub memory_total_mb: u64,
    pub memory_used_mb: u64,
    pub memory_free_mb: u64,
    pub disk_total_gb: u64,
    pub disk_used_gb: u64,
    pub disk_avail_gb: u64,
    pub disk_used_pct: u32,
    pub uptime_hours: f64,
    pub crontab_active: bool,
    pub telegram_bot_active: bool,
    pub sidecar_active: bool,
    pub last_batch_run: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LogResponse {
    pub log_name: String,
    pub lines: Vec<String>,
}
